sympy/concrete/summations.py

```diff
@@ -991,19 +991,21 @@ def eval_sum_symbolic(f, limits):
         r = gosper_sum(f, (i, a, b))
 
         # to find Sum at points where denominator approaches to zero
+        from sympy.utilities.iterables import subsets
         if isinstance(r, (Mul,Add)):
             from sympy import ordered
-            var = r.free_symbols
-            var = [x for x in var if x not in limits]
+            var = [x for x in r.free_symbols if x not in limits]
             den = denom(together(r))
             if any(x in den.free_symbols for x in var):
                 cond = []
-                for v in ordered(var):
+                for v in ordered(set(var) & den.free_symbols):
```

Comment:
OK, since there is only 1 den, there is only a need to solve for a single variable -- all others are just permutations of the others, e.g. if den is x - y having condition Eq(x, y) is the same as having Eq(y, x). So how about

```python
non_limit = r.free_symbols - Tuple(*limits[1:]).free_symbols
den_sym = non_limit & den.free_symbols
for v in ordered(den_sym):
    try:
        s = solve(den, v)
        m = Eq(v, s[0])
        if m:
            args.append((Sum(f_orig.subs(*m.args), limits).doit(), m))
        else:
            pass  # for coverage -- will m ever be False?
        break
    except NotImplementedError:
        continue
args.append((r, True))
return Piecewise(*args)
```

Reply:
@smichr isn't it better to get  two different arguments : `Eq(x, y/(y + 1))` and `Eq(y, -x/(x - 1))`, in comparison of getting just one of them? Is it worth the overhead caused?

Reply:
I don't think there is any extra information in the second equality. If the first is true the second is necessarily true. But looking at what you wrote it would make sense to say `And(Eq(x, y/(y + 1)), Ne(y, -1))`.

Reply:
wouldn't it complicate things? I mean suppose there are four symbols in `den_sym` then we have to see for singularity at various combinations of other three symbols.